SAPHana/bin/SAPHanaSR-manageAttr

```diff
@@ -670,31 +680,110 @@ chk_multi_target_requirements() {
     return $chk
 }
 
+# get resource name from cib
+get_res() {
+    local resType=$1
+    cib=/var/lib/pacemaker/cib/cib.xml
+    # get resources for given $RSID
+    resS="$(grep '<nvpair name="SID"' $cib | sed -n "/value=\"${RSID}\"/s/.*id=\"\(.*\)-instance_attributes-.*>/\1/gp")"
+    if [ -n "$INO" ]; then
+        # get resources for given $INO
+        resI="$(grep '<nvpair name="InstanceNumber"' $cib | sed -n "/value=\"${INO}\"/s/.*id=\"\(.*\)-instance_attributes-.*>/\1/gp")"
```

Comment:
Same a above

Reply:
Is really cool :) You use sed even more advanced then I can do that :)
However I do not know, if the id names really ALWAYS are formatted that way.

Reply:
First we might need to identyfy IDs for primitives but we could not be sure about the order of id=, class= and type:

> <primitive id="rsc_SAPHanaCon_HA1_HDB00"  ... class="ocf" ... type="SAPHanaController"

Now we need to scan the parameters introduced by:

> <instance_attributes ....

Last we need to identify the correct parameter (here SID or InstanceNumber. Again we could not say anything about the oder of name=, value= and if=.

> <nvpair name="SID" value="HA1" id="rsc_SAPHanaCon_HA1_HDB00-instance_attributes-SID"

I am not sure, if the id of a nvpair MUST match the id of the resource (here primitive).